diff --git a/manifests/package.pp b/manifests/package.pp
index ba8f6f7..cedb66e 100644
--- a/manifests/package.pp
+++ b/manifests/package.pp
@@ -1,4 +1,11 @@
-define r::package($r_path = '', $repo = 'http://cran.rstudio.com', $dependencies = false, $timeout = 300) {
+define r::package($r_path = '', $repo = 'http://cran.rstudio.com', $dependencies = false, $timeout = 300, $local = false) {
+
+    if $local == true {
+      $repo = NULL;
+    }
+    else {
+      $repo = "'${repo}'";
+    }
 
     case $::osfamily {
     '^(Debian|RedHat)$': {
@@ -12,8 +19,8 @@ define r::package($r_path = '', $repo = 'http://cran.rstudio.com', $dependencies
       }
 
       $command = $dependencies ? {
-        true    => "${binary} -e \"install.packages('${name}', repos='${repo}', dependencies = TRUE)\"",
-        default => "${binary} -e \"install.packages('${name}', repos='${repo}', dependencies = FALSE)\""
+        true    => "${binary} -e \"install.packages('${name}', repos=${repo}, dependencies = TRUE)\"",
+        default => "${binary} -e \"install.packages('${name}', repos=${repo}, dependencies = FALSE)\""
       }
 
       exec { "install_r_package_${name}":
